src/object_store.cpp

```diff
@@ -517,6 +517,19 @@ void ObjectStore::verify_compatible_for_read_only(std::vector<SchemaChange> cons
     verify_no_errors<InvalidSchemaChangeException>(verifier, changes);
 }
 
+void ObjectStore::verify_compatible_for_read_only_alternative(std::vector<SchemaChange> const& changes)
+{
+    using namespace schema_change;
+    struct Verifier : SchemaDifferenceExplainer {
+        using SchemaDifferenceExplainer::operator();
+
+        void operator()(RemoveProperty) { }
```

Comment:
Is it intentional that `AddTable` / `AddInitialProperties` are not handled here like they are for immutable mode? If I understand correctly, that will result in an error if the local schema contains any classes that aren't present in the file being opened. That would appear to preclude opening a synced Realm that doesn't already exist locally using this new read-only mode, since we'll think the schema isn't compatible until after the Realm has been downloaded.

With the immutable schema mode, missing tables are simply treated as if they were empty. That seems like it should also work for this mode.